containers/Dockerfile.buildenv

```diff
@@ -124,24 +235,34 @@ RUN apt-get update -y \
     && printf "if [ -f /etc/bash_completion ] && ! shopt -oq posix; then\n\
     . /etc/bash_completion\nfi\n\n" >> /root/.bashrc
 
-
+# WARNING: We switch to an empty container briefly!!!
+#
 # Cross-compile dependencies on the host system that take a long time to build
 # when emulating the target architecture. The `xbuild` stage is an image with
 # the same architecture as the host. It compiles the packages for the target
 # architecture. Then we can copy the compiled packages into the `base` image.
+#
+# Note: because we only cross-compile the latest Ubuntu version we support,
+# we just delete the executables on Ubuntu 18.04 and disable doxygen and ccache.
 FROM --platform=$BUILDPLATFORM tonistiigi/xx AS xx
 FROM --platform=$BUILDPLATFORM ubuntu:22.04 AS xbuild
 WORKDIR /root
 COPY --from=xx / /
 ARG TARGETPLATFORM
+ARG UBUNTU_VERSION=22.04
 ARG PARALLEL_MAKE_ARG=-j4
+
 # Install build dependencies for the host system
 RUN apt-get update -y \
-    && apt-get install -y cmake clang lld wget
+    && apt-get install -y cmake clang lld flex bison wget git
```

Comment:
Drop the changes until line 303? No changes to the xbuild stage should be necessary

Reply:
No, these were necessary to get cross compiling to work.